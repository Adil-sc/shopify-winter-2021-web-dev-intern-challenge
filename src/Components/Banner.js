import React, { useState } from 'react'
import Confetti from 'react-confetti'

export const Banner = (props) => {
  const [confettiIsFalling, setConfettiIsFalling] = useState(true)

  setTimeout(() => {
    setConfettiIsFalling(false)
  }, 5000)

  return (
    <div>
      <div className="h-14 bg-green-300 shadow rounded-md text-center">
        <p className="text-lg font-bold p-2">
          {`ðŸŽ‰ Congratulations! You've nominated at least 5 movies! ðŸŽ‰`}
        </p>
        {confettiIsFalling && <Confetti numberOfPieces={200} />}
      </div>
    </div>
  )
}

export default Banner
