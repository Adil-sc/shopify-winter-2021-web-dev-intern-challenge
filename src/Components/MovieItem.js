import React, { useContext, useState, useEffect } from 'react'
import { NominationContext } from '../Context/NominationContext'
import { addNominationToLocalStorage } from '../Utils/localStorage'
import {
  nominateButtonEnabled,
  nominateButtonDisabled,
} from '../Constants/cssConstants'

export const MovieItem = (props) => {
  const { nominations, setNominations } = useContext(NominationContext)
  const [isNominateButtonDisabled, setIsNominateButtonDisabled] = useState(
    false
  )

  useEffect(() => {
    if (nominations.get(props.movie.imdbID)) {
      setIsNominateButtonDisabled(true)
    } else {
      setIsNominateButtonDisabled(false)
    }
  }, [nominations, props.movie.imdbID])

  const addToNominations = (k, v) => {
    setNominations(new Map(nominations.set(k, v)))
    addNominationToLocalStorage(k, JSON.stringify(v))
    setIsNominateButtonDisabled(true)
  }

  return (
    <div className="pt-2 flex justify-center">
      <div className="max-w-sm w-full lg:max-w-full lg:flex shadow-lg mb-2 sm:mb-4">
        <div
          className="h-48 lg:h-auto lg:w-48 flex-none bg-cover rounded-t lg:rounded-t-none lg:rounded-l text-center overflow-hidden object-cover"
          style={{
            //Can add alt background via second url param: , url(https://image.flaticon.com/icons/svg/49/49238.svg)
            backgroundImage: `url(${props.movie.Poster})`,
            backgroundColor: '#9ae6b4',
          }}
        ></div>

        <div className="pl-4 w-full  bg-white rounded-b lg:rounded-b-none lg:rounded-r  flex flex-col justify-between leading-normal">
          <div className="mb-4">
            <div
              id="movieTitle"
              className="text-gray-900 font-bold text-xl mb-2"
            >
              {props.movie.Title}
            </div>
            <p className="text-xs text-gray-500 mb-4">
              Release Date: {props.movie.Year}
            </p>

            <div className="flex justify-end mr-4">
              <button
                className={
                  isNominateButtonDisabled
                    ? nominateButtonDisabled
                    : nominateButtonEnabled
                }
                disabled={isNominateButtonDisabled}
                onClick={() => {
                  addToNominations(props.movie.imdbID, props.movie)
                }}
              >
                Nominate
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default MovieItem
