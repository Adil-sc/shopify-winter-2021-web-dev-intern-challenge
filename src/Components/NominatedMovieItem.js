import React, { useContext } from 'react'
import { NominationContext } from '../Context/NominationContext'
import { removeNominationFromLocalStorage } from '../Utils/localStorage'

export const NominatedMovieItem = (props) => {
  const { nominations, setNominations } = useContext(NominationContext)

  const removeNominatedMovie = (k) => {
    if (nominations.get(k)) {
      removeNominationFromLocalStorage(k)
      nominations.delete(k)
      setNominations(new Map(nominations))
    }
  }

  return (
    <div className="pt-2 flex justify-center">
      <div className="max-w-sm w-full lg:max-w-full lg:flex shadow-lg mb-2">
        <div
          className="h-48 lg:h-auto lg:w-48 flex-none bg-cover rounded-t lg:rounded-t-none lg:rounded-l text-center overflow-hidden object-cover"
          style={{
            backgroundImage: `url(${props.movie.Poster})`,
            backgroundColor: '#9ae6b4',
          }}
        ></div>

        <div className="pl-4 w-full  bg-white rounded-b lg:rounded-b-none lg:rounded-r  flex flex-col justify-between leading-normal">
          <div className="mb-4">
            <div
              id="nominatedMovieTitle"
              className="text-gray-900 font-bold text-xl mb-2"
            >
              {props.movie.Title}
            </div>
            <p className="text-xs text-gray-500 mb-4">
              Release Date: {props.movie.Year}
            </p>

            <div className="flex justify-end mr-4">
              <button
                className="ml-4 bg-red-500 hover:bg-red-700 \
              text-white font-bold  px-4 rounded-full focus:outline-none"
                onClick={() => {
                  removeNominatedMovie(props.movie.imdbID)
                }}
              >
                Remove
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default NominatedMovieItem
