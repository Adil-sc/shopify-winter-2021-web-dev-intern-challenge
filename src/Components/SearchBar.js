import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'

const SearchBar = (props) => {
  return (
    <div className="h-auto  bg-white shadow rounded-md">
      <div className="p-4">
        <p className="text-sm pb-1 text-gray-700">Movie Title</p>

        <div className="flex items-center">
          <span className="absolute pl-3 ">
            <FontAwesomeIcon icon={faSearch} />
          </span>
          <input
            className="shadow appearance-none border px-10 rounded w-full p-2 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="username"
            type="text"
            placeholder="Search movies..."
            onChange={(e) => props.onSearchChange(e.target.value.trim())}
          ></input>
        </div>
      </div>
    </div>
  )
}

export default SearchBar
