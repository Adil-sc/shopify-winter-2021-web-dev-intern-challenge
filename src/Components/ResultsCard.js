import React from 'react'
import MovieItem from './MovieItem'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEllipsisH } from '@fortawesome/free-solid-svg-icons'

export const ResultsCard = (props) => {
  const renderResultsList = () => {
    return props.searchQuery.movies.map((movie) => (
      <div key={movie.imdbID}>
        <MovieItem movie={movie} />
      </div>
    ))
  }

  const renderEmptyState = () => {
    return (
      <div>
        <p className=" text-gray-500">
          Start searching to find a movie to nominate
        </p>
      </div>
    )
  }

  const renderAnimation = () => {
    return (
      <div className="animate-ping h-5 w-5">
        <span className="pl-2 ">
          <FontAwesomeIcon icon={faEllipsisH} />
        </span>
      </div>
    )
  }

  const noResultsFound = () => {
    return (
      <div className="text-gray-700 ">
        {props.searchQuery.searchText &&
        props.searchQuery.movies == null &&
        !props.handleLoading.isLoading
          ? 'No movies found. Please try another search'
          : null}
      </div>
    )
  }

  return (
    <div className="h-auto w-full bg-white shadow rounded-md p-8 mt-6 grid col-span-1 sm:mb-4">
      <h2 className="text-2xl font-bold mb-2">{`Results ${
        props.searchQuery.searchText
          ? `for  "${props.searchQuery.searchText}"`
          : ''
      }`}</h2>

      <div id="displayResults" className="">
        {props.searchQuery.movies ? renderResultsList() : renderEmptyState()}
        {props.handleLoading.isLoading ? renderAnimation() : null}
      </div>
      {noResultsFound()}
    </div>
  )
}

export default ResultsCard
