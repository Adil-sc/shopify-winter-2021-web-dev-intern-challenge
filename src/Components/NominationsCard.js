import React from 'react'
import NominatedMovieItem from './NominatedMovieItem'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCheck } from '@fortawesome/free-solid-svg-icons'

export const NominationsCard = (props) => {
  const nominations = props.nominations
  const renderNominationsList = () => {
    return [...nominations.keys()].map((nomination) => {
      const movie = nominations.get(nomination)
      return <NominatedMovieItem key={movie.imdbID} movie={movie} />
    })
  }

  const renderEmptyState = () => {
    return (
      <div>
        <p className=" text-gray-500">Your nominated movies appear here</p>
      </div>
    )
  }

  const savedToDeviceIndicator = () => {
    return (
      <p className="text-xs text-gray-500 ml-4 sm:ml-0">
        <span>
          <FontAwesomeIcon icon={faCheck} />{' '}
        </span>
        Saved to device
      </p>
    )
  }

  return (
    <div className="h-auto full bg-white shadow rounded-md p-8 mt-6">
      <div className="flex justify-between">
        <h2 className="text-2xl font-bold mb-2">Nominations</h2>
        {nominations.size > 0 ? savedToDeviceIndicator() : null}
      </div>

      <div id="displayNominations">
        {nominations.size ? renderNominationsList() : renderEmptyState()}
      </div>
    </div>
  )
}

export default NominationsCard
