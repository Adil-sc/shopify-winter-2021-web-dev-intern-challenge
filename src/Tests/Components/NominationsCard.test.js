import React from 'react'
import { shallow } from 'enzyme'
import { NominationsCard } from '../../Components/NominationsCard'
import { nominations, emptyNominations } from '../Fixtures/Movies'

describe('<NominationsCard />', () => {
  it('should render empty NominationsCard correctly', () => {
    const wrapper = shallow(<NominationsCard nominations={emptyNominations} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should display <NominatedMovieItem /> if nominations exist', () => {
    const wrapper = shallow(<NominationsCard nominations={nominations} />)
    expect(wrapper.find('#displayNominations').text()).toBe(
      '<NominatedMovieItem />'
    )
  })
})
