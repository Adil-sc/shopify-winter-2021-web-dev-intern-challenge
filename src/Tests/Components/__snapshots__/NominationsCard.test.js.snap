// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NominationsCard /> should render empty NominationsCard correctly 1`] = `
<div
  className="h-auto full bg-white shadow rounded-md p-8 mt-6"
>
  <div
    className="flex justify-between"
  >
    <h2
      className="text-2xl font-bold mb-2"
    >
      Nominations
    </h2>
  </div>
  <div
    id="displayNominations"
  >
    <div>
      <p
        className=" text-gray-500"
      >
        Your nominated movies appear here
      </p>
    </div>
  </div>
</div>
`;
